src/gui/spatialIndicator.js

```diff
@@ -0,0 +1,419 @@
+                cachedWorldObject = null; // don't accept the local world object
+            }
+            if (cachedWorldObject && !cachedOcclusionObject) {
+                cachedOcclusionObject = realityEditor.gui.threejsScene.getObjectForWorldRaycasts(cachedWorldObject.objectId);
+                if (cachedOcclusionObject) {
+                    // trigger the callback and clear the interval
+                    callback(cachedWorldObject, cachedOcclusionObject);
+                    clearInterval(occlusionDownloadInterval);
+                    occlusionDownloadInterval = null;
+                }
+            }
+        }, 1000);
+    }
+
+    let worldIntersectPoint = {};
+    function getRaycastCoordinates(screenX, screenY) {
+        // todo: get objectsToCheck outside of this function & stop resetting & pushing every time this function runs,
+        // todo: in order to make the calculation more efficient
+        let objectsToCheck = [];
+        if (cachedOcclusionObject) {
+            objectsToCheck.push(cachedOcclusionObject);
+        }
+        objectsToCheck = objectsToCheck.concat(indicatorList);
+        // if (realityEditor.gui.threejsScene.getGroundPlaneCollider()) {
+        //     objectsToCheck.push(realityEditor.gui.threejsScene.getGroundPlaneCollider());
+        // }
+        if (cachedWorldObject && objectsToCheck.length > 0) {
+            // by default, three.js raycast returns coordinates in the top-level scene coordinate system
+            let raycastIntersects = realityEditor.gui.threejsScene.getRaycastIntersects(screenX, screenY, objectsToCheck);
+            if (raycastIntersects.length > 0) {
+                // if hit a cylinder indicator, then make the cylinders expand and last longer
+                if (raycastIntersects[0].object.parent.name === 'cylinderIndicator') {
+                    console.log('spatial cursor hit a cylinder indicator in the scene!');
+                    raycastIntersects[0].object.parent.iclick++;
+                    return;
+                }
+                let groundPlaneMatrix = realityEditor.sceneGraph.getGroundPlaneNode().worldMatrix;
+                let inverseGroundPlaneMatrix = new realityEditor.gui.threejsScene.THREE.Matrix4();
+                realityEditor.gui.threejsScene.setMatrixFromArray(inverseGroundPlaneMatrix, groundPlaneMatrix);
+                inverseGroundPlaneMatrix.invert();
+                raycastIntersects[0].point.applyMatrix4(inverseGroundPlaneMatrix);
+                let trInvGroundPlaneMat = inverseGroundPlaneMatrix.clone().transpose();
+                worldIntersectPoint = {
+                    point: raycastIntersects[0].point,
+                    normalVector: raycastIntersects[0].face.normal.clone().applyMatrix4(trInvGroundPlaneMat).normalize(),
+                }
+            }
+        }
+        return worldIntersectPoint; // these are relative to the world object
+    }
```

Comment:
I was thinking it's probably better to export this function at the bottom of spatialCursor.js rather than redefine it here. You can give the function there an optional third parameter that lets you customize which objectsToCheck. But I see you're doing some custom logic on the result based on what it hits, so refactoring would become more complex.